SDWebImage/SDImageCache.m

```diff
@@ -126,17 +121,19 @@ - (nullable NSString *)cachePathForKey:(nullable NSString *)key {
     return [self.diskCache cachePathForKey:key];
 }
 
-- (nullable NSString *)makeDiskCachePath:(nonnull NSString *)fullNamespace {
+- (nullable NSString *)userCacheDirectory {
     NSArray<NSString *> *paths = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES);
-    return [paths.firstObject stringByAppendingPathComponent:fullNamespace];
+    return paths.firstObject;
 }
 
 - (void)migrateDiskCacheDirectory {
     if ([self.diskCache isKindOfClass:[SDDiskCache class]]) {
         static dispatch_once_t onceToken;
         dispatch_once(&onceToken, ^{
-            NSString *newDefaultPath = [[self makeDiskCachePath:@"default"] stringByAppendingPathComponent:@"com.hackemist.SDImageCache.default"];
-            NSString *oldDefaultPath = [[self makeDiskCachePath:@"default"] stringByAppendingPathComponent:@"com.hackemist.SDWebImageCache.default"];
+            // ~/Library/Caches/com.hackemist.SDImageCache/default/
+            NSString *newDefaultPath = [[[self userCacheDirectory] stringByAppendingPathComponent:@"com.hackemist.SDImageCache"] stringByAppendingPathComponent:@"default"];
+            // ~/Library/Caches/default/com.hackemist.SDImageCache.default/
+            NSString *oldDefaultPath = [[[self userCacheDirectory] stringByAppendingPathComponent:@"default"] stringByAppendingPathComponent:@"com.hackemist.SDWebImageCache.default"];
```

Comment:
How many users is using `5.x` beta? this change the behavior for `5.x` beta which already released previously. If many people already used it, the migrate operation also need to contain `5.x beta` to this new directory.

Reply:
No, what I mentioned is not `API breaking`, like change API or delete API, it's the changes of internal cache directory. I don't find `CocoaPods` statistics, see the `Carthage` [statistics](https://libraries.io/carthage/rs%2FSDWebImage/usage), 4.48% users choose `5.x`, we need responsible for them, they already do directory transfer from `4.x`. After this change, it would leads images lost and wasted disk space.

Reply:
I decide to not do migration for those folder. Because migration it's not a cheap. And it's not a good idea to do any tie dependency of the cache folder. Because it's can become invalid at any time. Apple iOS system may purge your User Cache folder as well.

Expect this issue, any extra consideration about this PR ?

Reply:
? It's easy to fix, check `5.x` released beta directory wether existed, if exist, `4.x`'s directory already transferred, so `oldDefaultPath` is `5.x` released beta directory. If `5.x` released beta directory not exist, the `oldDefaultPath` is `4.x` directory.

Reply:
@skyline75489 TBO, I'm ok if you guys think user need to get punishment when he use `beta` 😂 . NIT, I suppose give some hints that tell user we change the directory again.

Reply:
I wouldn't say it's "punishment". We can not make it better without breaking things. Beta is where things will be broken. And that's a good thing cuz that's also where it gets better.

Reply:
@dreampiggy I'm not found any other issue except this transfer. I think We can merge if you guys think it's acceptable.

@skyline75489 I used the emoji after I said "punishment", means it's just a joke. 😢

Reply:
It's quite simple to do this migration for 5.0.0-beta3 user, just a simpe line of code to call. But I'm afraid this migration list will become more and more huge, like a pain.

If I'm a author of a K-V storage or database framework, I'll be carefully about the migration. But actually, SD's cache is a simple cache, no important user files are here, and it's always purgable.

I can add one **Important Change** section in the Changelog or migration guide, to point this out about directory during 5.x beta versions. People can also call that migration code by their own just as we do for them, actually.

Reply:
@dreampiggy 👌 We give the reminder in the docs, it may acceptable.